lib/Transport.js

```diff
@@ -386,7 +386,15 @@ class Transport {
 
       debug('Sniffing ended successfully', result.body)
       const protocol = result.meta.connection.url.protocol || 'http:'
-      const hosts = this.connectionPool.nodesToHost(result.body.nodes, protocol)
+      var url = ''
+      if (Object.prototype.hasOwnProperty.call(result.meta.connection, 'url') &&
+          typeof result.meta.connection.url === 'string') {
```

Comment:
This can be:

```js
if (typeof result.meta.connection.url === "string") {
  // ...
}
```

If it is not there, then it is `undefined`, therefore not a `string`.